evalml/pipelines/pipeline_base.py

```diff
@@ -425,6 +426,60 @@ def feature_importance(self):
         df = pd.DataFrame(importance, columns=["feature", "importance"])
         return df
 
+    def graph_json(self):
+        """Generates a JSON with nodes consisting of the component names and parameters, and edges detailing component relationships.
+
+        Returns:
+            dag_json (str): A serialized JSON representation of a DAG structure.
+        """
```

Comment:
The explanation/template in the description for this PR was really helpful. Could you add that to the docs somewhere, either here in the docstring or somewhere in the User Guide?

Reply:
I'll add it in both!